cluster/config.go

```diff
@@ -140,6 +140,19 @@ func BuildContainerConfig(c container.Config, h container.HostConfig, n network.
 		}
 	}
 
+	// If a user specified a Swarm classic-style network name in the
+	// endpoints config (i.e. <node name>/<network name>), strip off the
+	// node name so that the daemon will recognize the network.
+	endpointsConfig := map[string]*network.EndpointSettings{}
+	for networkName, v := range n.EndpointsConfig {
+		parts := strings.SplitN(networkName, "/", 2)
+		if len(parts) > 1 {
+			networkName = parts[1]
```

Comment:
I think this we'll be a backward breaking change, as I can create custom networks having a `/`;

```
 docker network create foo/bar/baz

docker network ls

NETWORK ID          NAME                DRIVER              SCOPE
5ce5654a460e        bridge              bridge              local
81d2f269d9ff        docker_gwbridge     bridge              local
7296e31838b7        foo/bar/baz         bridge              local
26d3887604d2        host                host                local
59e5e41d3a7b        none                null                local
```

Reply:
Hmm, I guess we'll have to iterate through and see if `parts[0]` is the name of a node.